kartothek/core/common_metadata.py

```diff
@@ -78,6 +79,36 @@ def _schema_compat(self):
 
             schema = schema.remove_metadata()
             md = {b"pandas": _dict_to_binary(pandas_metadata)}
+            # https://github.com/JDASoftwareGroup/kartothek/issues/259
+
+            if schema is not None:
+                fields = []
+                for f in schema:
+                    if pa.types.is_timestamp(f.type):
+                        if ARROW_LARGER_EQ_0150:
+                            f = pa.field(f.name, pa.timestamp("us", tz=f.type.tz))
+                        else:
+                            ## using pandas meta to pass timezone
+                            # pandas_field = [field for field in pandas_metadata['columns'] if field['name']==f.name]
+                            pandas_field = list(
+                                filter(
+                                    lambda x: x["name"] == f.name,
+                                    pandas_metadata["columns"],
+                                )
+                            )
```

Comment:
No need for lambdas and iterator->list conversions here:

```suggestion
                            pandas_field = [
                                col
                                for col in pandas_metadata["columns"]
                                if col["name"] == f.name
                            ]
```